### PR TITLE
Better ThreadUtil.sleep, return boolean and interrupt current thread

### PR DESCRIPTION
**Pull request by DanVanAtta**
Instead of swallowing the interrupted exception we will instead return false. This will allow clients to control the behavior as they would like. As per best practices, we also now interrupt the current thread. When catching an InterruptedException, the thing that read if the thread was interrupted, will have reset the interrupted flag. So to handle that it is best practice to set the interrupted flag again.

Further background and discussion:  https://github.com/triplea-game/triplea/issues/1448

### CODE REVIEW

File: src/games/strategy/triplea/ui/TripleAFrame.java [hunk 1]
```diff
@@ -1474,24 +1474,25 @@ public void requiredTurnSeries(final PlayerID player) {
...
-      }
-      // center on capital of player, if it is a new player
-      if (!player.equals(lastStepPlayer)) {
-        lastStepPlayer = player;
-        data.acquireReadLock();
-        try {
-          mapPanel.centerOn(TerritoryAttachment.getFirstOwnedCapitalOrFirstUnownedCapital(player, data));
-        } finally {
-          data.releaseReadLock();
+    if(ThreadUtil.sleep(300)) {
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/ui/TripleAFrame.java]: Would it be better to just return from the method if return false rather than nesting this?

File: src/games/strategy/triplea/pbem/AxisAndAlliesForumPoster.java [hunk 2]
```diff
@@ -182,7 +182,9 @@ public boolean postTurnSummary(final String message, final String subject) {
               + m_topicId + ".0;num_replies=" + numReplies);
           post.addRequestHeader("Accept", "*/*");
           // the site has spam prevention which means you can't post until 15 seconds after login
-          ThreadUtil.sleep(15 * 1000);
+          if(ThreadUtil.sleep(15 * 1000)) {
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/pbem/AxisAndAlliesForumPoster.java]: Shouldn't this check for false instead of true?

File: src/games/strategy/engine/message/UnifiedMessengerHub.java [hunk 3]
```diff
@@ -144,7 +144,9 @@ private void invoke(final HubInvoke hubInvoke, final Collection<INode> remote, f
   public void waitForNodesToImplement(final String endPointName) {
     final long endTime = NODE_IMPLEMENTATION_TIMEOUT + System.currentTimeMillis();
     while (System.currentTimeMillis() < endTime && !hasImplementors(endPointName)) {
-      ThreadUtil.sleep(50);
+      if(!ThreadUtil.sleep(50)) {
+        break;
```
Comment (ron-murhammer) [inline src/games/strategy/engine/message/UnifiedMessengerHub.java]: Better to have a return instead of break here? Functionally equivalent I believe.

File: src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java [hunk 4]
```diff
@@ -179,7 +179,6 @@ public String remoteShutdown(final String hashedPassword, final String salt) {
     if (encryptedPassword.equals(hashedPassword)) {
       (new Thread(() -> {
         System.out.println("Remote Shutdown Initiated.");
-        ThreadUtil.sleep(1000);
```
Comment (ron-murhammer) [inline src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: Why was this removed?

Comment (RoiEXLab) [inline src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: And if this is being removed on purpose, the thread running this code is senseless

File: src/games/strategy/engine/message/UnifiedMessengerHub.java (hunk 3 above)
Comment (RoiEXLab) [inline src/games/strategy/engine/message/UnifiedMessengerHub.java]: I agree!

**Comment by RoiEXLab** [comment]
This Travis Build status, not changing to green like it should is triggering me >:(

File: src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java (hunk 4 above)
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: It was intentional. The sleep here is nonsensical. Rather than ignore the return value and have a pointless sleep, it was removed.

**Comment by DanVanAtta** [comment]
Updated

File: src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java (hunk 4 above)
Comment (ron-murhammer) [inline src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: @DanVanAtta What is the point of a new Thread if you are just going to immediately call System.exit(0)? My guess is this was to give something 1 second to complete. No idea if that makes any real sense but if we are gonna remove the sleep then shouldn't we remove the new thread as well?

Comment (DanVanAtta) [inline src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: @ron-murhammer I'm not sure if it is actually safe to remove the thread. I welcome that as a follow on task. It's quite clear we do not need the sleep.

**Comment by DanVanAtta** [comment]
@ron-murhammer I'm not willing to accept the scope creep. Removing the thread where we had an unnecessary sleep is not the current focus.
